test/common/router/config_impl_test.cc

```diff
@@ -2382,6 +2382,16 @@ class RouterMatcherHashPolicyTest : public testing::Test, public ConfigImplTestB
     return *config_;
   }
 
+  absl::optional<uint64_t> generateHash(const std::vector<absl::string_view>& header_values) {
+    Http::TestRequestHeaderMapImpl headers = genHeaders("www.lyft.com", "/foo", "GET");
+    for (auto& value : header_values) {
+      headers.addCopy("foo_header", std::string(value));
```

Comment:
nit: you don't need to make a std::string for value; you can pass it directly to addCopy,  whose signature is:
```
  virtual void addCopy(const LowerCaseString& key, absl::string_view value) PURE;
```

Reply:
The `TestRequestHeaderMapImpl` only provides the std::string one, https://github.com/envoyproxy/envoy/blob/main/test/test_common/utility.h#L905

Reply:
I think the string_view one is there: 
https://github.com/envoyproxy/envoy/blob/87a588e929adf7280c313507a2dad1bb60551893/test/test_common/utility.h#L945

It has to be in order to implement the class interface.

Reply:
emm... I can't compile it. I have to make the key as LowerCaseString, since it requires explicit conversion.